zhaquirks/xiaomi/aqara/motion_agl1.py

```diff
@@ -0,0 +1,179 @@
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+from zhaquirks.xiaomi import XiaomiAqaraE1Cluster
+
+APPROACH_DISTANCE_ATTR_ID = 0x015B  # UINT32 The configurable maximum detection distance in millimeters (default 600 = 6 meters).
+MOTION_ATTR_ID = 0x0160  # UINT8 Detected motion (0x02 = no movement, 0x03 = large movement, 0x04 = small movement)
+MOTION_DISTANCE_ATTR_ID = 0x015F  # UINT32 Distance to the detected motion (mm)
+MOTION_SENSITIVITY_ATTR_ID = 0x010C  # UINT8 The configurable detection sensitivity (0x01 = low, 0x02 = medium, 0x03 = high, default 0x03 = High)
+OCCUPANCY_ATTR_ID = 0x0142  # UINT8 Occupancy detected 0x0 = absence, 0x01 = presence
+RESET_NO_PRESENCE_STATUS_ATTR_ID = 0x0157  # UINT8 Trigger AI spatial learning
+RESTART_DEVICE_ATTR_ID = 0x00E8  # BOOL Trigger device restart
+
+RESET_NO_PRESENCE_STATUS_WRITE_VALUE = 1
+RESTART_DEVICE_WRITE_VALUE = 0
+
+
+class AqaraMotionSensitivity(types.enum8):
+    """Aqara motion sensitivity."""
+
+    Low = 0x01
+    Medium = 0x02
+    High = 0x03
+
+
+class AqaraMotion(types.enum8):
+    """Aqara motion."""
+
+    Unknown_0 = 0x00
+    Unknown_1 = 0x01
+    Idle = 0x02
+    Moving = 0x03
+    Still = 0x04
+
+
+class OppleCluster(XiaomiAqaraE1Cluster):
+    """Aqara manufacturer cluster for the FP1E presence sensor."""
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Manufacturer specific attributes."""
+
+        approach_distance = ZCLAttributeDef(
+            id=APPROACH_DISTANCE_ATTR_ID,
+            type=types.uint32_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        motion = ZCLAttributeDef(
+            id=MOTION_ATTR_ID,
+            type=types.uint8_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        motion_distance = ZCLAttributeDef(
+            id=MOTION_DISTANCE_ATTR_ID,
+            type=types.uint32_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        motion_sensitivity = ZCLAttributeDef(
+            id=MOTION_SENSITIVITY_ATTR_ID,
+            type=types.uint8_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        occupancy = ZCLAttributeDef(
+            id=OCCUPANCY_ATTR_ID,
+            type=types.uint8_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        reset_no_presence_status = ZCLAttributeDef(
+            id=RESET_NO_PRESENCE_STATUS_ATTR_ID,
+            type=types.uint8_t,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+        restart_device = ZCLAttributeDef(
+            id=RESTART_DEVICE_ATTR_ID,
+            type=types.Bool,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+    def _update_attribute(self, attrid: int, value: Any) -> None:
+        super()._update_attribute(attrid, value)
+        if attrid == OCCUPANCY_ATTR_ID:
```

Comment:
This would be more consistent with our other (new) quirks:
```suggestion
        if attrid == self.AttributeDefs.occupancy.id:
```

Reply:
Thanks for the feedback @TheJulianJES , updated with https://github.com/zigpy/zha-device-handlers/pull/3521/commits/0636e87c22e7c247ad57ac9c690953e511745f72 to follow this recommendation.